src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1353,6 +1366,94 @@ mod tests {
         )
     }
 
+    #[test]
+    fn tdl_arg_expr_group() -> IonResult<()> {
+        eval_template_invocation(
+            r#"(macro abc () (make_string (; "a" "b" "c")))"#,
+            r#"
+                (:abc)
+            "#,
+            r#"
+                "abc"
+            "#,
+        )
+    }
+
+    #[test]
+    fn multiple_arg_expr_groups() -> IonResult<()> {
+        stream_eq(
+            r#"
+                $ion_encoding::(
+                    (macro_table
+                        (macro foo ($x+ $y* $z+)
+                            (make_string (; $x "-" $y "-" $z)))
+                        (macro letters ()
+                            (foo
+                                (; "a" "b" "c")
+                                (; "d" "e" "f")
+                                (; "g" "h" "i")))
+                    )
+                )
+                (:letters)
+            "#,
+            r#"
+                "abc-def-ghi"
+            "#,
+        )
+    }
+
+    #[test]
+    fn explicit_expr_group_arg() -> IonResult<()> {
+        stream_eq(
+            r#"
+            (:append_macros
+                (macro greet ($x) (make_string (; "Hello, " $x)))
+            )
+            (:greet "Gary")
+        "#,
+            r#"
+            "Hello, Gary"
+        "#,
+        )
+    }
+
+    #[test]
+    fn built_in_append_macros() -> IonResult<()> {
+        stream_eq(
+            r#"
```

Comment:
Can you add a symbol table at the start (and reference it later) to make sure that `append_macros` doesn't clobber the symbol table?